framework/src/media/audio/resample/samplerate.c

```diff
@@ -458,46 +470,53 @@ int src_simple(src_handle_t handle, src_data_t *src_data)
 		// To be exact, both origin_sample_rate & desired_sample_rate shouldn't be changed.
 		// Same restriction for sample width and channel num.
 		RETURN_VAL_IF_FAIL((src->lastratio == src_ratio), SRC_ERR_NOT_SUPPORT);
-		RETURN_VAL_IF_FAIL((src->channels_num == channels_num), SRC_ERR_NOT_SUPPORT);
+		RETURN_VAL_IF_FAIL((src->channels_num == new_channel_num), SRC_ERR_NOT_SUPPORT);
 		RETURN_VAL_IF_FAIL((src->lastoldformat == old_sample_width), SRC_ERR_NOT_SUPPORT);
 		RETURN_VAL_IF_FAIL((src->lastnewformat == new_sample_width), SRC_ERR_NOT_SUPPORT);
 	}
 
 	int input_frames_used = 0;
 	int output_frames_gen = 0;
+
 	if (old_sample_rate == new_sample_rate) {
-		// same rate, just copy data
-		input_frames_used = MINIMUM(src_data->input_frames, src->outsize_in_frames);
-		output_frames_gen = input_frames_used;
-		memcpy((void *)src->out_buffer,\
-			   src_data->data_in,\
-			   output_frames_gen * channels_num * bytes_per_sample);
+		// Same sample rate
+		int32_t ret;
+		ret = rechannel(ch2layout(old_channel_num), ch2layout(new_channel_num),\
+				(const int16_t *)src_data->data_in, src_data->input_frames,\
+				(int16_t *)src->out_buffer, src->out_buffer_frames);
+		RETURN_VAL_IF_FAIL((ret > 0), SRC_ERR_BAD_PARAMS);
+		output_frames_gen = ret;
+		input_frames_used = output_frames_gen;
 	} else {
-		int frames_num = 0; // frames count in internal fin buffer
+		int frames_num = 0; // number of frames in internal input buffer
 
-#define FRAMES_TO_BYTES(frames) ((frames) * channels_num * bytes_per_sample)
+#define NEW_FRAMES_TO_BYTES(frames) ((frames) * new_channel_num * bytes_per_sample)
```

Comment:
please move this define to top area, it is our coding rule.

Reply:
Ok, I will fix it